src/Airship/Types.hs

```diff
@@ -217,6 +228,29 @@ eitherResponse reqDate reqParams dispatched req resource = do
     (e, trace) <- runWebmachine reqDate reqParams dispatched req resource
     return (both e, trace)
 
+errorResponse :: Monad m =>
+                 UTCTime
+              -> Request
+              -> ErrorResponses m
+              -> m Response
```

Comment:
I'm not sure but I suspect if you make this function return `Webmachine m Response` you can remove some of the duplication with `resourceToWaitT` (eg. `runRWST`) down below.